src/backend/distributed/executor/multi_utility.c

```diff
@@ -1385,32 +1420,71 @@ PlanRenameStmt(RenameStmt *renameStmt, const char *renameCommand)
 	 * RenameRelation(), renameatt() and RenameConstraint(). However, since all
 	 * three statements have identical lock levels, we just use a single statement.
 	 */
-	relationId = RangeVarGetRelid(renameStmt->relation, AccessExclusiveLock,
-								  renameStmt->missing_ok);
+	objectRelationId = RangeVarGetRelid(renameStmt->relation,
+										AccessExclusiveLock,
+										renameStmt->missing_ok);
 
 	/*
 	 * If the table does not exist, don't do anything here to allow PostgreSQL
 	 * to throw the appropriate error or notice message later.
 	 */
-	if (!OidIsValid(relationId))
+	if (!OidIsValid(objectRelationId))
 	{
 		return NIL;
 	}
 
 	/* we have no planning to do unless the table is distributed */
-	isDistributedRelation = IsDistributedTable(relationId);
+	switch (renameStmt->renameType)
+	{
+		case OBJECT_TABLE:
+		case OBJECT_COLUMN:
+		case OBJECT_TABCONSTRAINT:
+		{
+			/* the target object is our tblRelationId. */
+			tableRelationId = objectRelationId;
+			break;
+		}
+
+		case OBJECT_INDEX:
+		{
+			/*
+			 * here, objRelationId points to the index relation entry, and we
+			 * are interested into the entry of the table on which the index is
+			 * defined.
+			 */
+			tableRelationId = IndexGetRelation(objectRelationId, false);
+			break;
+		}
+
+		default:
+
+			/*
+			 * Nodes that are not supported by Citus: we pass-through to the
+			 * main PostgreSQL executor. Any Citus-supported RenameStmt
+			 * renameType must appear above in the switch, explicitely.
+			 */
+			return NIL;
+	}
+
+	isDistributedRelation = IsDistributedTable(tableRelationId);
+
```

Comment:
super minor style/consistency thingy: we usually don't have an empty line between the `if` statement and the result it checks.

Reply:
ok